lib/compat/wordpress-6.3/class-gutenberg-rest-global-styles-controller-6-3.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * REST API: Gutenberg_REST_Global_Styles_Controller class
+ *
+ * @package    Gutenberg
+ * @subpackage REST_API
+ */
+
+/**
+ * Base Global Styles REST API Controller.
+ */
+class Gutenberg_REST_Global_Styles_Controller_6_3 extends Gutenberg_REST_Global_Styles_Controller_6_2 {
```

Comment:
Could we not just use `WP_REST_Revisions_Controller`?

Reply:
My understanding is that, Gutenberg, I need to inherit [Gutenberg_REST_Global_Styles_Controller_6_2](https://github.com/WordPress/gutenberg/blob/ac0dc37f4c51fbeba889994c9b44cca8cff548a0/lib/compat/wordpress-6.2/class-gutenberg-rest-global-styles-controller-6-2.php#L12) to ensure functionality works for folks who haven't upgraded their version of WordPress yet.

It's a pattern that been around for a while, e.g., 
https://github.com/WordPress/gutenberg/blob/ac0dc37f4c51fbeba889994c9b44cca8cff548a0/lib/compat/wordpress-6.3/class-gutenberg-rest-pattern-directory-controller-6-3.php#L12-L12

I'm adding a new route `'/wp/v2/global-styles/revisions'` to the [existing endpoint Gutenberg_REST_Global_Styles_Controller](https://github.com/ramonjd/wordpress-develop/blob/b24f990f4f7789cee65628ef090a9e68a2798b5b/src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php#L87-L87) (`'/wp/v2/global-styles'`).

Furthermore I'm adding a new link, `version-history`, to the response of `'/wp/v2/global-styles'`.

Do you think I should create a new controller for global styles revisions?

Reply:
> Do you think I should create a new controller for global styles revisions?

Hmm, I to think perhaps I should create `Gutenberg_REST_Global_Styles_Revisions_Controller` so that I can create methods consistent with WP_Rest patterns as you've pointed out, and, where possible, the response from `WP_REST_Revisions_Controller` as well (see below)

<img width="874" alt="Screenshot 2023-04-24 at 12 08 45 pm" src="https://user-images.githubusercontent.com/6458278/233884105-a7522723-1ea5-4c32-8900-17f1a1544771.png">


Global styles are a bit "special" since the expected response is JSON so there's no `content.rendered`.

I'll play with it anyway 😄 🍺